cypress/support/api_commands.js

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getRandomInt} from '../utils';
+
+import users from '../fixtures/users.json';
+import theme from '../fixtures/theme.json';
+
+// *****************************************************************************
+// Read more:
+// - https://on.cypress.io/custom-commands on writing Cypress commands
+// - https://api.mattermost.com/ for Mattermost API reference
+// *****************************************************************************
+
+// *****************************************************************************
+// Authentication
+// https://api.mattermost.com/#tag/authentication
+// *****************************************************************************
+
+/**
+ * Login a user directly via API
+ * @param {String} username - e.g. "user-1" (default)
+ */
+Cypress.Commands.add('apiLogin', (username = 'user-1') => {
+    const user = users[username];
+
+    cy.request({
+        url: '/api/v4/users/login',
+        method: 'POST',
+        body: {login_id: user.username, password: user.password},
+    });
+});
+
+/**
+ * Logout a user directly via API
+ */
+Cypress.Commands.add('apiLogout', () => {
+    cy.request({
+        url: '/api/v4/users/logout',
+        method: 'POST',
+    });
+});
+
+// *****************************************************************************
+// Channels
+// https://api.mattermost.com/#tag/channels
+// *****************************************************************************
+
+/**
+ * Creates a channel directly via API
+ * This API assume that the user is logged in and has cookie to access
+ * @param {String} teamId - The team ID of the team to create the channel on
+ * @param {String} name - The unique handle for the channel, will be present in the channel URL
+ * @param {String} displayName - The non-unique UI name for the channel
+ * @param {String} type - 'O' for a public channel (default), 'P' for a private channel
+ * @param {String} purpose - A short description of the purpose of the channel
+ * @param {String} header - Markdown-formatted text to display in the header of the channel
+ * All parameters required except purpose and header
+ */
+Cypress.Commands.add('apiCreateChannel', (teamId, name, displayName, type = 'O', purpose = '', header = '') => {
+    const uniqueName = `${name}-${getRandomInt(9999).toString()}`;
+
+    return cy.request({
+        headers: {'X-Requested-With': 'XMLHttpRequest'},
+        url: '/api/v4/channels',
+        method: 'POST',
+        body: {
+            team_id: teamId,
+            name: uniqueName,
+            display_name: displayName,
+            type,
+            purpose,
+            header,
+        },
+    }).then((response) => {
```

Comment:
Is it necessary to explicitly yield with `then(res => res)`?

Reply:
Oh, nice catch!  I initially added that to log what's being returned from that API.  Forgot to remove it, thanks!
Please take a look again.